paasta_tools/vitesscluster_tools.py

```diff
@@ -15,6 +17,594 @@
+                "mountPath": "/etc/vault/all_cas",
+                "name": "vault-secrets",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "/etc/vitess_keyspace_acls",
+                "name": "acls",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "etc/credentials.yaml",
+                "name": "vttablet-fake-credentials",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "/etc/init_db.sql",
+                "name": "keyspace-fake-init-script",
+                "readOnly": True,
+            },
+        ],
+        "extraVolumes": [
+            {"name": "vault-secrets", "hostPath": {"path": "/nail/etc/vault/all_cas"}},
+            {
+                "name": "acls",
+                "hostPath": {"path": "/nail/srv/configs/vitess_keyspace_acls"},
+            },
+            {"name": "vttablet-fake-credentials", "hostPath": {"path": "/dev/null"}},
+            {"name": "keyspace-fake-init-script", "hostPath": {"path": "/dev/null"}},
+        ],
+        "replicas": 1,
+        "vttablet": {
+            "extraFlags": vttablet_extra_flags,
+            "resources": {
+                "requests": {
+                    "cpu": "100m",
+                    "memory": "256Mi",
+                },
+                "limits": {"memory": "256Mi"},
+            },
+        },
+        "externalDatastore": {
+            "database": db_name,
+            "host": SOURCE_DB_HOST,
+            "port": port,
+            "user": "vt_app",
+            "credentialsSecret": {
+                "key": "/etc/credentials.yaml",
+                "volumeName": "vttablet-fake-credentials",
+            },
+        },
+        "dataVolumeClaimTemplate": {
+            "accessModes": ["ReadWriteOnce"],
+            "resources": {"requests": {"storage": "10Gi"}},
+            "storageClassName": "ebs-csi-gp3",
+        },
+    }
+
+    vttablet_extra_env = copy.deepcopy(VTTABLET_EXTRA_ENV)
+    for env in vttablet_extra_env:
+        if env["name"] == "TOPOLOGY_FLAGS":
+            env[
+                "value"
+            ] = f"--topo_implementation {TOPO_IMPLEMENTATION} --topo_global_server_address ${zk_address} --topo_global_root {TOPO_GLOBAL_ROOT}"
+        if env["name"] == "CELL_TOPOLOGY_SERVERS":
+            env["value"] = zk_address
+        if env["name"] == "DB":
+            env["value"] = db_name
+        if env["name"] == "KEYSPACE":
+            env["value"] = keyspace
+        if env["name"] == "VAULT_ADDR":
+            env["value"] = f"https://vault-dre.{region}.yelpcorp.com:8200"
+        if env["name"] == "VAULT_CACERT":
+            env["value"] = f"/etc/vault/all_cas/acm-privateca-{region}.crt"
+        config["extraEnv"].append(env)
+
+    # Add extra pod label to filter
+    config["extraLabels"]["tablet_type"] = f"{db_name}_{tablet_type}"
+
+    return config
+
+
+def build_keyspaces_config(
```

Comment:
same as above, we'll want to add types here

Reply:
https://github.com/Yelp/paasta/pull/3889/commits/c7a6b7f7665b11eb4e812cba4a46457b851040cd